### DISCUSSION TITLE
Unable to publish new meeting -- pending and draft are only options

### DISCUSSION THREAD

**Discussion by rockridgefellowship** [code4recovery/12-step-meeting-list#812] [Q&A] [answered]
Hi there --

Using Wordpress 6, and TSML 3.14.3, I'm unable to publish a new meeting -- I can only save as Draft or as Pending Review (there is no Published option).

All other areas of site are working as expected, as are previously saved meetings (they have Published as an option).

Ideas?

Thanks,

Patrick

**Comment by joshreisner** [comment, marked as answer]
hi @rockridgefellowship sorry to hear about it. sounds like a geocoding issue to me. what value is selected on your meetings > import & settings > settings page for "Address Geocoding?"

when editing a meeting with the publish button greyed out, could you open the [javascript console](https://webmasters.stackexchange.com/questions/8525/how-do-i-open-the-javascript-console-in-different-browsers) and show any errors or messages you see there?

thanks

**Reply by rockridgefellowship** [reply]
Hey @joshreisner ! Sorry for the delay -- Address GeoCoding is using the legacy method; We have a Mapbox token.

I noticed today that a meeting I try to edit keeps changing "Can I currently attend this meeting in person?" back to "yes". And, good tip, here's some errors from the console when I tick that button to "no":

`11:15:13.362 Uncaught TypeError: can't convert undefined to object
    setMapMarkers https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    createMap https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    <anonymous> https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    jQuery 8
    <anonymous> https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    jQuery 2
admin.min.js:1:647207
    setMapMarkers https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    createMap https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    <anonymous> https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    jQuery 8
    <anonymous> https://rockridgefellowship.org/wp-content/plugins/12-step-meeting-list/assets/js/admin.min.js?ver=3.14.3:1
    jQuery 2
`

**Reply by joshreisner** [reply]
hmmm i cant seem to recreate this on a test site, so it sounds like it's having trouble geocoding the address. what's the meeting address? is it the same one you use for all your meetings? `3989 Howe St, Oakland, CA 94611, USA`? strange that it would have a problem here but not on the other listings.

here's a test meeting i created at rockridge with the legacy method and no errors: https://code4recovery.org/meetings/test-rockridge

what happens if you create a new online meeting at this location? does it run into the same issue?

**Reply by foundartfactory** [reply]
Howdy! Weird -- looks great on your site :) ...The address you have is correct, and yes, Any new online meeting runs into this issue

**Reply by joshreisner** [reply]
hm, well it's hard to know really what's happening here. but if my hunch is right and something has gone awry with geocoding then perhaps overriding that address with a hard-coded value would fix it.

do you have access to your site's `wp-config.php`? try adding this code at the bottom:

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '3989 Howe St, Oakland, CA 94611, USA => array(
            'formatted_address' => '3989 Howe St, Oakland, CA 94611, USA',
            'city' => 'Oakland',
            'latitude' => 37.8264368,
            'longitude' => -122.2545915,
            'approximate' => 'no',
        ),
    ));
}
```

you could also put it in your theme's `functions.php`

**Reply by foundartfactory** [reply]
Will do!, and thanks -- I'm using a child them, so I'll put it in the `functions.php` there for starters.

**Reply by taylord2020** [reply]
Hey Josh,

I'm having the same issue on the SantaBarbaraAA.com website. The last two meetings created will not publish, they hang with Status: Draft. Not sure if I should create a new thread. 

Thank you,

Taylor D

![image](https://user-images.githubusercontent.com/62483893/173887789-a302d88b-256a-445a-a683-da161508e8e7.png)


--

![image](https://user-images.githubusercontent.com/62483893/173887406-56d4de90-7d5b-4677-bd9f-79a6cd976501.png)

---

**Reply by joshreisner** [reply]
curious to hear if the hard-coded geocoding solution helped above.

@taylord2020 regarding your meeting, this could be a different issue. does your meeting have a valid address? if you are able to publish but that has no effect then it sounds like you need to give it an address. online meetings often get a general one, like `Santa Barbara, CA, USA`

**Reply by rockridgefellowship** [reply]
Howdy @joshreisner et al -- I added the code you posted to our child theme's functions.php, but no change. Thoughts?

**Reply by rockridgefellowship** [reply]
Just to clarify; this is what I'm experiencing (attached), whether for in-person or online meetings. 
<img width="814" alt="Screen Shot 2022-06-17 at 08 05 13" src="https://user-images.githubusercontent.com/101430516/174325274-f5737323-9f08-441a-9866-57f2ac566bfe.png">

**Reply by joshreisner** [reply]
hmm, if you hit the blue "Publish" button then what happens? It goes back to draft?

**Reply by joshreisner** [reply]
if i create a new entry in wordpress i would not expect "Published" to be an option in that dropdown until I hit the Publish button.

**Reply by rockridgefellowship** [reply]
Yes, when I hit the Publish button, it stays at Draft status.

**Reply by joshreisner** [reply]
so strange. could i see a screenshot of the location box on the page?

**Reply by rockridgefellowship** [reply]
So, I was going to  gomake a screenshot for you, and things seem to suddenly be working as expected. Consider me confused.

<img width="949" alt="Screen Shot 2022-06-17 at 18 27 01" src="https://user-images.githubusercontent.com/101430516/174417230-75f066fb-9a95-424b-9c4a-c853b2876d5a.png">

**Reply by joshreisner** [reply]
ok! well we do [have a little code](https://github.com/code4recovery/12-step-meeting-list/blob/main/includes/save.php#L303) that catches meetings without an address and saves them as a draft. if you run into that again, be sure that there is an address specified. (this might seem counterintuitive on a site where every meeting is at the same address.)

**Reply by rockridgefellowship** [reply]
OK! Thanks for that tip -- Will keep an eye on things, especially since I'm in the confused state 
(not working-->suddenly working). Curious if @taylord2020 has had any change?

**Reply by taylord2020** [reply]
@joshreisner

That suggestion worked (the address was missing). We are now able to publish meetings that were stuck in Draft mode. Going forward, for online-only meetings we will use our Central Office mailing address.

Thanks!

**Reply by joshreisner** [reply]
@taylord2020 great! 

FYI, the general service office recommends using a general location like `Santa Barbara, CA` for online meetings to avoid confusion

**Reply by taylord2020** [reply]
> @taylord2020 great!
> 
> FYI, the general service office recommends using a general location like `Santa Barbara, CA` for online meetings to avoid confusion

For some reason it wouldn't work without a full street address. I'll keep messing around with it though.
